src/PlexShareApp/MainScreenView.xaml.cs

```diff
@@ -32,42 +32,52 @@ public partial class MainScreenView : Window
         private static DashboardPage dashboardPage;
         private static WhiteBoardPage whiteBoardPage;
         private static ChatPageView chatPage;
-        private static ScreenSharePage screenSharePage;
+        private static ScreenshareServerView screenshareServerView;
+        private static ScreenshareClientView screenshareClientView;
         public event PropertyChangingEventHandler? PropertyChanged;
+        private bool isServer;
         
         public MainScreenView(string name, string email, string picPath, string url, string ip, string port)
         {
-            bool verified = true;
-
+            bool verified = false;
             IUXServerSessionManager serverSessionManager = SessionManagerFactory.GetServerSessionManager();
-            IUXClientSessionManager clientSessionManafer = SessionManagerFactory.GetClientSessionManager();
+            IUXClientSessionManager clientSessionManager = SessionManagerFactory.GetClientSessionManager();
 
             if (ip == "-1")
             {
                 MeetingCredentials meetingCredentials = serverSessionManager.GetPortsAndIPAddress();
-                verified = clientSessionManafer.AddClient(meetingCredentials.ipAddress, meetingCredentials.port, name);
+                verified = clientSessionManager.AddClient(meetingCredentials.ipAddress, meetingCredentials.port, name);
                 ip = meetingCredentials.ipAddress;
                 port = meetingCredentials.port.ToString();
+                isServer = true;
             }
             else
             {
-                verified = clientSessionManafer.AddClient(ip, int.Parse(port), name);
+                verified = clientSessionManager.AddClient(ip, int.Parse(port), name);
+                isServer = false;
             }
 
             if (verified)
             {
-
                 InitializeComponent();
                 dashboardPage = new DashboardPage();
-                whiteBoardPage = new WhiteBoardPage();
                 chatPage = new ChatPageView();
-                screenSharePage = new ScreenSharePage();
+                if (isServer)
+                {
+                    whiteBoardPage = new WhiteBoardPage(0);
+                }
+                else
+                {
+                    whiteBoardPage = new WhiteBoardPage(1);
+                }
+                screenshareServerView = new ScreenshareServerView();
+                screenshareClientView = new ScreenshareClientView();
```

Comment:
Our code should be in the if condition as well. We want the server view to be made only when the machine is a server machine and the client view to be called only when it is a client machine